agent/stats.go

```diff
@@ -54,8 +55,18 @@ func (s *statsReporter) callback(_, _ time.Time, virtual, _ map[netlogtype.Conne
 	s.L.Lock()
 	defer s.L.Unlock()
 	s.logger.Debug(context.Background(), "got stats callback")
-	s.networkStats = &virtual
-	s.unreported = true
+	// Accumulate stats until they've been reported.
+	if s.unreported {
+		if s.networkStats == nil && virtual != nil {
+			s.networkStats = make(map[netlogtype.Connection]netlogtype.Counts)
```

Comment:
Nit: let's save some allocations.

```suggestion
			s.networkStats = make(map[netlogtype.Connection]netlogtype.Counts, len(virtual))
```

Reply:
I've never actually benchmarked how much a difference a size hint gives for maps, especially ones that don't have a lot of data. Is there a significant difference?

Your suggestion made me realize this had a better fix 😄.